### PR TITLE
rename nodeModulesPath to prevent conflict with ember-cli internals

### LINKED ISSUE #101: Build throws Cannot set property nodeModulesPath of #<Class> which has only a getter for >=0.1.10
Upgrading `@fortawesome/ember-fontawesome` from `0.1.9` to `0.1.10` or higher breaks build (and all other Ember CLI commands). It throws: `Cannot set property nodeModulesPath of #<Class> which has only a getter`

It seems to be related to https://github.com/FortAwesome/ember-fontawesome/commit/9eaa462a020ef98c4ebdd65c0874087f3c0745d9.

```
$ ember -v
ember-cli: 2.18.2
node: 8.15.1
os: linux x64
```

Since it may be related to `broccoli-rollup`:

```
$ yarn list --pattern broccoli-rollup
yarn list v1.15.2
├─ broccoli-rollup@2.1.1
└─ ember-data@2.16.3
   └─ broccoli-rollup@1.3.0
Done in 1.10s.
```

Error report by Ember CLI:

```
 VERSIONS:
  - ares: 1.10.1-DEV
  - cldr: 32.0
  - http_parser: 2.8.0
  - icu: 60.1
  - modules: 57
  - napi: 3
  - nghttp2: 1.33.0
  - node: 8.15.1
  - openssl: 1.0.2r
  - tz: 2017c
  - unicode: 10.0
  - uv: 1.23.2
  - v8: 6.2.414.75
  - zlib: 1.2.11

ERROR Summary:

  - broccoliBuilderErrorStack: [undefined]
  - code: [undefined]
  - codeFrame: [undefined]
  - errorMessage: Cannot set property nodeModulesPath of #<Class> which has only a getter
  - errorType: [undefined]
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
  - message: Cannot set property nodeModulesPath of #<Class> which has only a getter
  - name: TypeError
  - nodeAnnotation: [undefined]
  - nodeName: [undefined]
  - originalErrorMessage: [undefined]
  - stack: TypeError: Cannot set property nodeModulesPath of #<Class> which has only a getter
    at assignProperties (/home/main/Code/fey/pruefung-client/node_modules/core-object/lib/assign-properties.js:68:19)
    at Function.extend (/home/main/Code/fey/pruefung-client/node_modules/core-object/core-object.js:43:7)
    at Function.Addon.lookup (/home/main/Code/fey/pruefung-client/node_modules/ember-cli/lib/models/addon.js:1765:27)
    at graph.each (/home/main/Code/fey/pruefung-client/node_modules/ember-cli/lib/models/addons-factory.js:54:38)
    at Vertices.each (/home/main/Code/fey/pruefung-client/node_modules/dag-map/dag-map.umd.js:197:13)
    at Vertices.walk (/home/main/Code/fey/pruefung-client/node_modules/dag-map/dag-map.umd.js:125:14)
    at DAG.each (/home/main/Code/fey/pruefung-client/node_modules/dag-map/dag-map.umd.js:68:24)
    at AddonsFactory.initializeAddons (/home/main/Code/fey/pruefung-client/node_modules/ember-cli/lib/models/addons-factory.js:45:11)
    at Project.initializeAddons (/home/main/Code/fey/pruefung-client/node_modules/ember-cli/lib/models/project.js:442:38)
    at Project.eachAddonCommand (/home/main/Code/fey/pruefung-client/node_modules/ember-cli/lib/models/project.js:504:12)
```

### PR DESCRIPTION
**Pull request by jelhan**
Ember CLI < 3.0 has an internal property `nodeModulesPath`. It was deprecated and removed cause it _"is not a reliable way to discover npm modules. Instead, consider doing: require("resolve").sync(something, { basedir: project.root })."_ So we might want to change to mechanism here also.

Fixes #101

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,11 +17,11 @@ const findWorkspaceRoot = require('find-yarn-workspace-root');
 module.exports = {
   name: require('./package').name,
   fontawesomeConfig: null,
-  nodeModulesPath: null,
+  _nodeModulesPath: null,
 
   treeForVendor(vendorTree) {
     const iconRollups = []
-    const pathToCore = path.join(this.nodeModulesPath, '@fortawesome', 'fontawesome-svg-core');
+    const pathToCore = path.join(this._nodeModulesPath, '@fortawesome', 'fontawesome-svg-core');
```
Comment (jelhan) [inline index.js]: Using `resolve.sync()` as recommended in deprecation message would look like this:

```js
const pathToCore = require('path').dirname(
  require('resolve').sync('@fortawesome/fontawesome-svg-core')
);
```

File: index.js [hunk 2]
```diff
@@ -49,7 +49,7 @@ module.exports = {
             resolve()
           ]
         },
-        nodeModulesPath: this.nodeModulesPath,
+        _nodeModulesPath: this._nodeModulesPath,
```
Comment (jrjohnson) [inline index.js]: This should be `nodeModulesPath: this._nodeModulesPath`. It is a parameter for Broccoli Rollup and they decide what it's called.

File: index.js [hunk 3]
```diff
@@ -70,7 +70,7 @@ module.exports = {
           resolve()
         ]
       },
-      nodeModulesPath: this.nodeModulesPath,
+      _nodeModulesPath: this._nodeModulesPath,
```
Comment (jrjohnson) [inline index.js]: This should be `nodeModulesPath: this._nodeModulesPath`.

File: index.js (hunk 1 above)
Comment (jrjohnson) [reply index.js]: This is a good change too @jelhan, do you want to go ahead and add it here? Or in a separate PR on this this is merged?

We've already imported resolve so it would probably be `path.dirname(resolve.sync('@fortawesome/fontawesome-svg-core'))`.

File: index.js (hunk 2 above)
Comment (jelhan) [reply index.js]: Good catch. :+1: Sometimes a find and replace isn't the safest way to do it. :smile:

File: index.js (hunk 1 above)
Comment (jelhan) [reply index.js]: This would be better in a separate PR. It should address all usages of `nodeModulesPath` in my opinion cause having different strategies would be even worse. Also I'm not aware of the edge cases in which a custom `nodeModulePath` must be provided to `Rollup`. Therefor I don't know how to test that changes manually.

**Comment by jelhan** [comment]
@jrjohnson Would be great if you could bundle a release with this fix.
